### DISCUSSION TITLE
Live video captured over Surfshark VPN cannot display content properly

### DISCUSSION THREAD

**Discussion by rickygai** [aclap-dev/video-downloadhelper#1104] [Help and Q&A]
### Your question

**Microsoft Windows Version 22H2 (OS Build 22621.3958)
Microsoft Edge Version 127.0.2651.86 (Official build) (64-bit)
Chrome Version 127.0.6533.89 (Official Build) (64-bit)
Video DownloadHelper for Edge Version 9.0.2.12
Surfshark VPN v5.8.2 ( Desktop Version )**

Upon connected to Surfshark VPN to watch and capture live channel, the videos captured under Microsoft Edge and Chrome browsers cannot show the contents properly when played, but audio is fine as shown below. 

![image](https://github.com/user-attachments/assets/20b40430-3fcf-4fd4-a8df-e0c320de9479)

Please advise or do something. Thank you.

### Addon & CoApp version and details

Video DownloadHelper for Edge Version 9.0.2.12
Companion App installed (v2.0.19)

**Comment by GrampaWildWilly** [comment]
You are showing a classic case of content that is protected by DRM.  It is easy in Firefox to tell whether content is protected by DRM.  That's the one browser you didn't try.  In Firefox, there are usually a few icons in the URL bar at the top of the browser window to the left of the URL.  You can hover the mouse over each icon & a tooltip hovers up.  If one of those tooltips mentions DRM, then the content is protected by DRM.  My understanding is that on the other 2 browsers, this feature is not available.  I'm going by what others have posted here.  I use neither of those browsers.

Your issue has nothing to do with your operating system, your browser, your version of VDH or the CoApp, your video player, or even your use of a VPN.  It has to do with protection by DRM.  Digital Rights Management.  DRM is a kludge that has been foisted on the world by a conspiracy of software & hardware developers in cahoots with content creators.  The intention is to protect  copyrighted material against unauthorized copying.  That is why VDH has never, does not now, nor will ever properly download content protected by DRM.  It is my understanding that it is possible to play content in the player on a web page because there is some magic combination of CPU circuitry, motherboard chipset circuitry, & video adapter circuitry that unscrambles any DRM content sent to them from the browser.  In other words, DRM is implemented on our computers in hardware.  There is something called the Widevine plugin that allows Firefox to pass DRM protected content to the hardware.  I have no idea how it's done in the other browsers.  Truth be told, I don't really understand it in Firefox, either.  I just see that this Widevine plugin comes embedded in Firefox & it permits me to play content protected by DRM.

It appears that something analagous to Widevine does not exist in the screen recorder niche.  I investigated screen recorders a few months ago & could not find a single one that would let me screen record livestreams protected by DRM.

I think DRM is being abused.  There are things that are protected by DRM that should not be.  For example, I used to be able to record livestreams of golf tournaments off Golf Channel.  But it's been over a year now since I've been able to do that because NBC Sports (owner of Golf Channel) has put all their content behind DRM protection.  I recorded the golf from the Tokyo Olympics 4 years ago.  It's not going to be possible this week.  My cable TV provider has a web interface that allows livestreaming of any channel in the lineup.  I think you won't be surprised to learn that it's all behind DRM protection.  Tennis Channel presents the same problem.  A number of our fellow users (I'm just a user, not an employee of VDH) have purchased online courses on Udemy & Hotmart.  They are encountering problems because haphazardly, some course material is behind DRM protection & some is not.  Some other fellow users are hitting the same issue on OnlyFans.

I say it's illegal.  There was a class action lawsuit about 50 or so years ago that obtained the decision that people could record TV shows for playback at a time more convenient to them than the networks' programming decision of when to air the content.  Time shifting.  This decision gave rise to the videocassette recorder industry, VHS VCRs.  I think this DRM business is the exact same issue.  But it's going to take another class action lawsuit to get this fixed.  There will be no end of squealing & squawking from those opposed to change.  It will drag on for years.  Lawyers on both sides will line their pockets.  And we'll probably all be long dead by the time any of this changes.

Color me optimistic.  Yeah right.

**Reply by GrampaWildWilly** [reply]
I clicked your link to CCTV.  That's the China national TV network, right?  It's all in Chinese & for some reason, Firefox didn't offer to translate the page for me, like I've seen on a number of other sites.  So I have no idea what I'm looking at.  I clicked a few links blindly, hoping to find something that might play.  But I didn't succeed.  I suppose I need a VPN to get in, same as you've done.  I'm not going to bother.  I'll leave it for somebody else to verify whether the content there is behind DRM protection.  Or maybe you could do it yourself.

**Reply by rickygai** [reply]
if it is under DRM protected, then just leave it will do, thanks anyway.
